# TASK AND SOLUTION (DAY-02)

### 1. Write a program that will give you in hand monthly salary after deduction on CTC - HRA(10%), DA(5%), PF(3%) and taxes deduction as below: Salary(Lakhs) : Tax(%)  Below 5 : 0%       5-10 : 10%       10-20 : 20%        aboove 20 : 30%

In [1]:
ctc = int(input('Enter your anual CTC:'))
if ctc < 5000000:
    salary=ctc*.82
elif ctc<1000000:
    salary=ctc*.72
elif ctc<2000000:
    salary=ctc*.62
else:
    salary=ctc*.52
print("You in hand monthly salary will be-", round(salary/12,2))

Enter your anual CTC:1200000
You in hand monthly salary will be- 82000.0


### 2. write e a program that take a user input of three angles and will find out whether it can form a triangle or not.

In [2]:
first = int(input('Enter the 1st angle:'))
second = int(input('Enter the 2nd angle:'))
third = int(input('Enter the 3rd angle:'))
if (first+second+third) == 180 and first>0 and second>0 and third>0:
  print('forms a triangle')
else:
  print('Does not form a triangle')

Enter the 1st angle:30
Enter the 2nd angle:30
Enter the 3rd angle:90
Does not form a triangle


### 3.Write a program that will take user input of cost price and selling price and determines whether its a loss or a profit.

In [3]:
cost_price = int(input('Enter cost price:'))
selling_price = int(input('Enter selling price:'))
if cost_price < selling_price:
    print('Profit')
elif cost_price > selling_price:
    print('Loss')
else:
    print('No Loss No Gain')

Enter cost price:120
Enter selling price:210
Profit


### 4.Write a menu-driven program - 1.cm to ft                     2.km to miles          3.USD to INR                                4.exit  Hint    1 cm = 0.032ft    1km = 0.62         1USD = 80 INR

In [4]:
menu = input("""
Hi select an option
1. cms to ft
2. km to miles
3. USD to INR
4. Exit
""")
if menu == '1':
  cm = float(input('Enter the cm value'))
  print('ft value is',0.032*cm)
elif menu == '2':
  km = float(input('Enter the km value'))
  print('miles value is',km*0.62)
elif menu == '3':
  usd = float(input('Enter usd'))
  print('inr',usd*80)
else:
  exit()


Hi select an option
1. cms to ft
2. km to miles
3. USD to INR
4. Exit
1
Enter the cm value12
ft value is 0.384


### 5.Display Fibonacci series up to 10 terms.Note: The Fibonacci Sequence is a series of numbers.
### The next number is found by adding up the two numbers before it. The first two numbers are 0 and 1. For example, 0, 1, 1, 2, 3, 5, 8, 13, 21. The next number in this series above is 13+21 = 34

In [5]:
num1,num2 = 0,1
for i in range(10):
  print(num1)
  next = num1 + num2
  num1 = num2
  num2 = next

0
1
1
2
3
5
8
13
21
34


### 6.Find the factorial of a given number.Write a program to use the loop to find the factorial of a given number. The factorial (symbol: !) means to multiply all whole numbers from the chosen number down to 1.
### For example: calculate the factorial of 5  5! = 5 × 4 × 3 × 2 × 1 = 120   o/p=120

In [6]:
num = int(input('Enter the number:'))
fact = 1
for i in range(1,num+1):
  fact = fact*i
print(fact)

Enter the number:5
120


### 7.Reverse a given integer number.  Example: Input: 76542         Output:24567

In [7]:
number = int(input('Enter the number:'))
rev = 0
while number>0:
  last = number%10
  rev = rev*10 + last
  number = number//10
print(rev)

Enter the number:123456
654321


### 8.Take a user input as integer N. Find out the sum from 1 to N. If any number if divisible by 5, then skip that number.And if the sum is greater than 300, don't need to calculate the sum further more. Print the final result. And don't use for loop to solve this problem.

In [8]:
N = int(input('Enter the number:'))
sum = 0
i = 1
while i < N+1:
  if i % 5 == 0:
    i+=1
    continue

  sum += i

  if sum > 300:
    sum = sum - i
    break

  i+=1
print(sum)

Enter the number:50
276


### 9.Write a program that keeps on accepting a number from the user until the user enters Zero. Display the sum and average of all the numbers.

In [9]:
sum = 0
count = 0
while True:
  num = int(input('Enter number:'))
  if num == 0:
    break
  sum = sum + num
  count = count + 1
print('sum',sum)
print('avg',sum/count)

Enter number:1
Enter number:2
Enter number:3
Enter number:4
Enter number:5
Enter number:6
Enter number:7
Enter number:0
sum 28
avg 4.0


### 10.Write a program which will find all such numbers which are divisible by 7 but are not a multiple of 5, between 2000 and 3200 (both included). The numbers obtained should be printed in a comma-separated sequence on a single line.

In [10]:
L = []
for i in range(2000,3201):
  if i % 7 == 0 and i % 5 != 0:
    L.append(str(i))
print(",".join(L))

2002,2009,2016,2023,2037,2044,2051,2058,2072,2079,2086,2093,2107,2114,2121,2128,2142,2149,2156,2163,2177,2184,2191,2198,2212,2219,2226,2233,2247,2254,2261,2268,2282,2289,2296,2303,2317,2324,2331,2338,2352,2359,2366,2373,2387,2394,2401,2408,2422,2429,2436,2443,2457,2464,2471,2478,2492,2499,2506,2513,2527,2534,2541,2548,2562,2569,2576,2583,2597,2604,2611,2618,2632,2639,2646,2653,2667,2674,2681,2688,2702,2709,2716,2723,2737,2744,2751,2758,2772,2779,2786,2793,2807,2814,2821,2828,2842,2849,2856,2863,2877,2884,2891,2898,2912,2919,2926,2933,2947,2954,2961,2968,2982,2989,2996,3003,3017,3024,3031,3038,3052,3059,3066,3073,3087,3094,3101,3108,3122,3129,3136,3143,3157,3164,3171,3178,3192,3199


### 11: A robot moves in a plane starting from the original point (0,0). The robot can move toward UP, DOWN, LEFT and RIGHT  with a given steps.The trace of robot movement is shown as the following:  UP 5   DOWN 3   LEFT 3  RIGHT 2  !

In [11]:
pos = [0,0]
while True:
  s = input('Enter the robot path')
  if s == '!':
    break
  direction = s.split()[0]
  steps = int(s.split()[1])

  if direction == 'UP':
    pos[1] = pos[1] + steps
  elif direction == 'DOWN':
    pos[1] = pos[1] - steps
  elif direction == 'LEFT':
    pos[0] = pos[0] - steps
  elif direction == 'RIGHT':
    pos[0] = pos[0] + steps
  else:
    pass
print('new pos',pos)
print((pos[0]**2 + pos[1]**2)**0.5)

Enter the robot path!
new pos [0, 0]
0.0


### 12.Write a program to print whether a given number is a prime number or not.

In [12]:
num = int(input('Enter the num:'))

flag = True
for i in range(2,num):
  if num%i == 0:
    flag = False
    break

if flag == True:
  print('Prime')
else:
  print('Not Prime')

Enter the num:21
Not Prime


### 13:Print all the Armstrong numbers in a given range. Range will be provided by the user Armstrong number is a number that is equal to the sum of cubes of its digits. For example 0, 1, 153, 370, 371 and 407  are the Armstrong numbers.

In [13]:
start = int(input("Enter the start of the range: "))
end = int(input("Enter the end of the range: "))

for num in range(start, end+1):
    temp = num
    sum = 0
    order = len(str(num))
    while temp > 0:
        digit = temp % 10
        sum += digit ** order
        temp //= 10
    if num == sum:
        print(num)

Enter the start of the range: 1
Enter the end of the range: 100
1
2
3
4
5
6
7
8
9


### 14. Calculate the angle between the hour hand and minute hand.
### Note: There can be two angles between hands; we need to print a minimum of two. Also, we need to print the floor of the final result angle. For example, if the final angle is 10.61, we need to print 10. Input: H = 9 , M = 0 Output: 90

In [14]:
h = int(input('Enter hours hand:'))
m = int(input('Enter minute hand:'))
# validate the input
if (h < 0 or m < 0 or h > 12 or m > 60):
    print('Wrong input')
# Idea is to minute angle - hour agnle from clockwise from 12 hour point
# 1 minute in minute angle manke 6 degree. (60 minute -> 360 degree)
m_angle = m*6
# every hour point yeilds to 30degree-- 12 hours-360degree plus if minute hand moves hour hands move too
# Every minute after hour hand take 0.5 degree movement. clockwise
h_angle = h*30 + m*0.5
# Take abs difference b/w them
angle = abs(h_angle - m_angle)
if angle>180:
    print(360-angle)
else:
    print(angle)

Enter hours hand:9
Enter minute hand:0
90.0


### 15.Given two rectangles, find if the given two rectangles overlap or not. A rectangle is denoted by providing the x and y coordinates of two points: the left top corner and the right bottom corner of the rectangle. Two rectangles sharing a side are considered overlapping. (L1 and R1 are the extreme points of the first rectangle and L2 and R2 are the extreme points of the second rectangle).
### Note: It may be assumed that the rectangles are parallel to the coordinate axis.

In [15]:
print("Enter both Rectangel points coordinates <x y>")
l1_x, l1_y = list(map(int, input('Enter L1 coordinate value  (Two value space seperated):').split()))
r1_x, r1_y = list(map(int, input('Enter r1 coordinate value,(Two value space seperated):').split()))

l2_x, l2_y = list(map(int, input('Enter L2 coordinate value,(Two value space seperated):').split()))
r2_x, r2_y = list(map(int, input('Enter r2 coordinate value,(Two value space seperated):').split()))

# if rectangle has area 0, no overlap
if l1_x == r1_x or l1_y == r1_y or r2_x == l2_x or l2_y == r2_y:
  print("Don't Overlap")
  # If one rectangle is on left side of other
elif l1_x > r2_x or l2_x > r1_x:
  print("Don't Overlap")

# If one rectangle is above other
elif r1_y > l2_y or r2_y > l1_y:
  print("Don't Overlap")
else:
  print("Overlap")

Enter both Rectangel points coordinates <x y>
Enter L1 coordinate value  (Two value space seperated):5 8
Enter r1 coordinate value,(Two value space seperated):8 7
Enter L2 coordinate value,(Two value space seperated):4 6
Enter r2 coordinate value,(Two value space seperated):7 9
Don't Overlap
